ngraph/core/shape_inference/include/space_to_batch_shape_inference.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstdint>
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/space_to_batch.hpp>
+#include <openvino/opsets/opset2.hpp>
+
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v1 {
+
+template <class T>
+void shape_infer(const ov::op::v1::SpaceToBatch* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    using ValType = typename std::iterator_traits<typename T::iterator>::value_type::value_type;
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 4 && output_shapes.size() == 1);
+
```

Comment:
Could you please add check 2-4 inputs here in the same way like you do it in BatchToSpace shape inference?

Reply:
Agreed. Will udpate